kubejs/server_scripts/base/recipes/powah/shaped.js

```diff
@@ -0,0 +1,206 @@
+        },
+        {
+            output: 'powah:player_transmitter_basic',
+            pattern: [' A ', 'BCB', ' D '],
+            key: {
+                A: 'powah:player_aerial_pearl',
+                B: 'powah:capacitor_basic_large',
+                C: 'powah:dielectric_casing',
+                D: 'powah:dielectric_rod'
+            },
+            id: `powah:crafting/player_tranmitter_basic`
+        },
+        {
+            output: 'powah:energy_hopper_basic',
+            pattern: ['AAA', 'BCB', 'ADA'],
+            key: {
+                A: 'powah:dielectric_paste',
+                B: 'powah:capacitor_basic_large',
+                C: 'powah:dielectric_casing',
+                D: 'minecraft:hopper'
+            },
+            id: `powah:crafting/energy_hopper_basic`
+        },
+        {
+            output: 'powah:energy_discharger_basic',
+            pattern: ['ABA', 'ACA', 'ABA'],
+            key: {
+                A: 'powah:dielectric_paste',
+                B: 'powah:capacitor_basic_large',
+                C: 'powah:dielectric_casing'
+            },
+            id: `powah:crafting/energy_discharger_basic`
+        }
+    ];
+
+    // powahTiers.forEach(function (tier) {
+    //     let capacitor = 'powah:capacitor_' + tier,
+    //         crystal = 'powah:crystal_' + tier,
+    //         cable = 'powah:energy_cable_' + tier;
+
+    //     if (tier == 'basic' || tier == 'starter') {
+    //         return;
+    //     } else if (tier == 'hardened') {
+    //         crystal = 'powah:steel_energized';
+    //     }
+
+    //     let lower_tiers = lowerTiers(powahTiers, tier);
+
+    //     recipes.push(
+    //         {
+    //             output: Item.of(`powah:energy_cell_${tier}`),
+    //             pattern: ['ABA', 'BCB', 'ABA'],
+    //             key: {
+    //                 A: crystal,
+    //                 B: capacitor,
+    //                 C: Ingredient.of(lower_tiers.map((item) => `powah:energy_cell_${item}`))
+    //             },
+    //             id: `${id_prefix}energy_cell_${tier}`
+    //         },
+    //         {
+    //             output: Item.of(`powah:battery_${tier}`),
+    //             pattern: [' A ', 'BCB', ' B '],
+    //             key: {
+    //                 A: crystal,
+    //                 B: capacitor,
+    //                 C: Ingredient.of(lower_tiers.map((item) => `powah:battery_${item}`))
+    //             },
+    //             id: `${id_prefix}battery_${tier}`
+    //         },
+
+    //         {
+    //             output: Item.of(`powah:solar_panel_${tier}`),
+    //             pattern: ['BCB', 'AAA'],
+    //             key: {
+    //                 A: crystal,
+    //                 B: capacitor,
+    //                 C: Ingredient.of(lower_tiers.map((item) => `powah:solar_panel_${item}`))
+    //             },
+    //             id: `${id_prefix}solar_panel_${tier}`
+    //         }
+    //     );
+    // });
```

Comment:
Could this be removed? (a lot of commented out code)

Reply:
Sure, thing.